components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1190,4 +1207,75 @@ double RewardsServiceImpl::GetContributionAmount() {
   return ledger_->GetContributionAmount();
 }
 
+void RewardsServiceImpl::FetchFavIcon(const std::string& url, const std::string& favicon_key) {
+  GURL parsedUrl(url);
+
+  if (!parsedUrl.is_valid()) {
+    return;
+  }
+
+  std::vector<std::string>::iterator it;
+  it = find (current_media_fetchers_.begin(), current_media_fetchers_.end(), url);
+  if (it != current_media_fetchers_.end()) {
+    VLOG(1) << "Already fetching favicon: " << url;
+    return;
+  }
+
+  current_media_fetchers_.emplace_back(url);
```

Comment:
`current_media_fetchers_` is also growing forever, pls clean it up after it's not needed.

Reply:
done